onnxscript/rewriter/llama_rule_sets_tests.py

```diff
@@ -0,0 +1,128 @@
+                    ],
+                    "name",
+                    [onnx.helper.make_tensor_value_info("X", FLOAT, [None, None, None])],
+                    [onnx.helper.make_tensor_value_info("Y", FLOAT, [None, None, None])],
+                ),
+                opset_imports=[onnx.helper.make_opsetid("", 18)],
+            ),
+            onnx.helper.make_model(
+                onnx.helper.make_graph(
+                    [
+                        onnx.helper.make_node("Mul", ["X", "one"], ["Y"]),
+                    ],
+                    "name",
+                    [onnx.helper.make_tensor_value_info("X", FLOAT, [None])],
+                    [onnx.helper.make_tensor_value_info("Y", FLOAT, [None])],
+                    [
+                        onnx.numpy_helper.from_array(
+                            np.array([1], dtype=np.float32), name="one"
+                        )
+                    ],
+                ),
+                opset_imports=[onnx.helper.make_opsetid("", 18)],
+            ),
+            onnx.helper.make_model(
+                onnx.helper.make_graph(
+                    [
+                        onnx.helper.make_node("Transpose", ["X"], ["xt"], perm=[1, 0]),
+                        onnx.helper.make_node("Transpose", ["xt"], ["Y"], perm=[1, 0]),
+                    ],
+                    "name",
+                    [onnx.helper.make_tensor_value_info("X", FLOAT, [None, None])],
+                    [onnx.helper.make_tensor_value_info("Y", FLOAT, [None, None])],
+                ),
+                opset_imports=[onnx.helper.make_opsetid("", 18)],
+            ),
+        ]
+        return models
+
+    def test_llama_p0_rule_set_identity(self):
+        for model in self._identity_models():
+            ir_model = ir.serde.deserialize_model(model)
+            rule_set = llama_rule_sets.llama_p0_rule_set()
+            rule_set.apply_to_model(ir_model)
+            rewritten_model = ir.serde.serialize_model(ir_model)
+
+            self.assertEqual(["Identity"], [n.op_type for n in rewritten_model.graph.node])
+            self._check_model(model, rewritten_model)
+
+    def _transpose_transpose_models(self):
+        models = [
+            onnx.helper.make_model(
+                onnx.helper.make_graph(
+                    [
+                        onnx.helper.make_node("Transpose", ["X"], ["xt"], perm=[1, 2, 0]),
+                        onnx.helper.make_node("Transpose", ["xt"], ["Y"], perm=[1, 2, 0]),
+                    ],
+                    "name",
+                    [onnx.helper.make_tensor_value_info("X", FLOAT, [None, None, None])],
+                    [onnx.helper.make_tensor_value_info("Y", FLOAT, [None, None, None])],
+                ),
+                opset_imports=[onnx.helper.make_opsetid("", 18)],
+            ),
+        ]
+        return models
+
+    def test_llama_p0_rule_set_transpose_transpose(self):
+        for model in self._transpose_transpose_models():
```

Comment:
Can this be parameterized so we can test all models in parallel?

Reply:
There are other tests. I guess the parallelism will apply at test level. I usually keep this syntax because when it fails, it does not run the other models. But the main reason is I prefer to keep the model close to the test because I don't have to look for it.